homeassistant/components/climate/broadlink.py

```diff
@@ -0,0 +1,447 @@
+    ip_addr = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+
+    if mac_addr is None:
+        _LOGGER.error("Wifi Thermostat: Invalid mac_addr !")
+        return False
+
+    if ip_addr is None:
+        _LOGGER.error("Wifi Thermostat: Invalid ip_addr !")
+        return False
+
+    if name is None:
+        _LOGGER.error("Wifi Thermostat: Invalid name !")
+        return False
+
+    wifi_thermostat = WifiThermostat(mac_addr,
+                        ip_addr,
+                        name,
+                        config.get(CONF_ADVANCED_CONFIG),
+                        config.get(CONF_SCHEDULE_WEEKDAY),
+                        config.get(CONF_SCHEDULE_WEEKDAY),
+                        config.get(CONF_MIN_TEMP),
+                        config.get(CONF_MAX_TEMP))
+
+    add_devices([BroadlinkThermostat(wifi_thermostat)], True)
+
+    def handle_set_schedule(service):
+        """Handle data for the set_schedule service call."""
+        thermostat = WifiThermostat(config.get(CONF_MAC),
+                                    config.get(CONF_HOST),
+                                    config.get(CONF_NAME),
+                                    config.get(CONF_ADVANCED_CONFIG),
+                                    config.get(CONF_SCHEDULE_WEEKDAY),
+                                    config.get(CONF_SCHEDULE_WEEKDAY),
+                                    config.get(CONF_MIN_TEMP),
+                                    config.get(CONF_MAX_TEMP))
+        schedule_wd = service.data.get(CONF_SCHEDULE_WEEKDAY)
+        schedule_we = service.data.get(CONF_SCHEDULE_WEEKEND)
+        thermostat.set_schedule(
+            {CONF_WEEKDAY:
+             json.loads(schedule_wd.replace("'", '"'), cls=Decoder),
+             CONF_WEEKEND:
+             json.loads(schedule_we.replace("'", '"'), cls=Decoder)})
+
+    #hass.services.register(DOMAIN, 'set_schedule', handle_set_schedule, schema=SET_SCHEDULE_SCHEMA)
+
+    def handle_set_advanced_conf(service):
+        """Handle data for the set_advanced_conf service call."""
+        thermostat = WifiThermostat(config.get(CONF_MAC),
+                                    config.get(CONF_HOST),
+                                    config.get(CONF_NAME),
+                                    config.get(CONF_ADVANCED_CONFIG),
+                                    config.get(CONF_SCHEDULE_WEEKDAY),
+                                    config.get(CONF_SCHEDULE_WEEKDAY),
+                                    config.get(CONF_MIN_TEMP),
+                                    config.get(CONF_MAX_TEMP))
+        advanced_conf = service.data.get(CONF_ADVANCED_CONFIG)
+        thermostat.set_advanced_config(
+            json.loads(advanced_conf.replace("'", '"'), cls=Decoder))
+
+    #hass.services.register(DOMAIN, 'set_advanced_conf', handle_set_advanced_conf, schema=SET_ADVANCED_CONF_SCHEMA)
+
+    _LOGGER.debug("Wifi Thermostat: Component successfully added !")
+
+
+class Decoder(json.JSONDecoder):
```

Comment:
Remove this. It shouldn't be part of this platform.

Reply:
where should I put it ?

Reply:
I don't think this should be included at all. Why is it needed?